src/Npgsql.NodaTime/Internal/PeriodConverter.cs

```diff
@@ -67,12 +67,20 @@ protected override void WriteCore(PgWriter writer, Period value)
         value = value.Normalize();
         // Note that the end result must be long
         // see #3438
-        var microsecondsInDay =
-            (((value.Hours * NodaConstants.MinutesPerHour + value.Minutes) * NodaConstants.SecondsPerMinute + value.Seconds) * NodaConstants.MillisecondsPerSecond + value.Milliseconds) * 1000 +
-            value.Nanoseconds / 1000; // Take the microseconds, discard the nanosecond remainder
+        long microsecondsInDay;
+        int days;
+        int months;
+        checked
+        {
+            microsecondsInDay =
+                (((value.Hours * NodaConstants.MinutesPerHour + value.Minutes) * NodaConstants.SecondsPerMinute + value.Seconds) * NodaConstants.MillisecondsPerSecond + value.Milliseconds) * 1000 +
+                value.Nanoseconds / 1000; // Take the microseconds, discard the nanosecond remainder
+            days = value.Weeks * 7 + value.Days;
+            months = value.Years * 12 + value.Months;
+        }
 
         writer.WriteInt64(microsecondsInDay);
-        writer.WriteInt32(value.Weeks * 7 + value.Days); // days
-        writer.WriteInt32(value.Years * 12 + value.Months); // months
+        writer.WriteInt32(days);
```

Comment:
Any reason to not just have the Write operations within the block above (at which point you don't need to separate the variable declarations)?

Reply:
I don't think `check { }` affects called code - in my mental model it only affects code directly (lexically) within the block...

Reply:
Looked at the docs and yeah, you're absolutely right. That's good to know. Though with how we now have a try/catch there we might as well just leave it as is.

Reply:
It seems like simpler code to just move it all inside the checked and try, but I don't really care - up to you.